cpp/src/copying/contiguous_split.cu

```diff
@@ -957,39 +1022,47 @@ std::vector<contiguous_split_result> contiguous_split(cudf::table_view const& in
 
   // copy.  1 block per buffer
   {
-    constexpr int block_size = 512;
-    copy_partition<<<num_bufs, block_size, 0, stream.value()>>>(
+    copy_partition<<<num_bufs, copy_block_size, 0, stream.value()>>>(
       num_src_bufs, num_partitions, d_src_bufs, d_dst_bufs, d_dst_buf_info);
   }
 
+  // DtoH dst info (to retrieve null counts)
+  CUDA_TRY(cudaMemcpyAsync(
+    h_dst_buf_info, d_dst_buf_info, dst_buf_info_size, cudaMemcpyDeviceToHost, stream.value()));
+
+  stream.synchronize();
```

Comment:
Is this intended?

Reply:
Yes.  Needed because build_output_columns() will need to read the computed null counts on the cpu.